src/libxtp/davidsonsolver.cc

```diff
@@ -315,49 +316,40 @@ Eigen::VectorXd DavidsonSolver::computeCorrectionVector(
     const Eigen::VectorXd &qj, double lambdaj,
     const Eigen::VectorXd &Aqj) const {
 
-  /* compute correction vector with either DPR or OLSEn CORRECTION
+  /* compute correction vector with either DPR or OLSEN CORRECTION
    * For details on the method see :
    * Systematic Study of Selected Diagonalization Methods
    * for Configuration Interaction Matrices
    * M.L. Leininger et al .
    * Journal of Computational Chemistry Vol 22, No. 13 1574-1589 (2001)
    */
 
-  Eigen::VectorXd out_vect;
   switch (this->_davidson_correction) {
     case CORR::DPR: {
-      out_vect = dpr(Aqj, lambdaj);
-      break;
+      return dpr(Aqj, lambdaj);
     }
-
     case CORR::OLSEN: {
-      out_vect = olsen(Aqj, qj, lambdaj);
-      break;
+      return olsen(Aqj, qj, lambdaj);
     }
   }
-  return out_vect;
+  return Eigen::VectorXd::Zero(0);
 }
 
 Eigen::VectorXd DavidsonSolver::dpr(const Eigen::VectorXd &r,
                                     double lambda) const {
-  /* \brief Compute the diagonal preconditoned residue : delta = - (D -
-   * lambda)^{-1} r
+  /* \brief Compute the diagonal preconditoned residue :
+  \delta = -r/(D - lambda)
    */
-  Eigen::VectorXd delta = r.array() / (lambda - _Adiag.array());
-  return delta;
+  return (-r.array() / (_Adiag.array() - lambda));
```

Comment:
why did you rewrite is like that ? Wouldn't it  be marginally more efficient the way it was ?

Reply:
THis way it is closer to the comment. I do not mind in the end.

Reply:
The optimised code will look the same most likely.